app/static/js/createEvents.js

```diff
@@ -36,128 +36,218 @@ function format24to12HourTime(timeStr) {
+      isEmpty = true;
+      $(eventNameInput).addClass('border-red');
+    } else{
+      $(eventNameInput).removeClass('border-red');
     }
-    else if(timeCheck){
-      $('#textNotifierPadding').addClass('pt-5');
-      $('.invalidFeedback').text("Event end time must be after start time");
-      $('.invalidFeedback').css('display', 'block');  
-      $('.invalidFeedback').on('animationend', function() {
-        $('.invalidFeedback').css('display', 'none');
-        $('#textNotifierPadding').removeClass('pt-5')
-      });
-      timeCheck= false;
-        
+  });
+
+  // Check if the date input field is empty
+  datePickerInputs.each((index, datePickerInput) => {
+    if (datePickerInput.value.trim() === '') {
+        isEmpty = true;
+        $(datePickerInput).addClass('border-red');
+    } else {
+      $(datePickerInput).removeClass('border-red');
     }
-    else {
-      storeMultipleOfferingEventAttributes();
-      pendingmultipleEvents = [];
-      $("#checkIsMultipleOffering").prop('checked', true);
-      // Remove the modal and overlay from the DOM
-      $('#modalMultipleOffering').modal('hide');
-      $('.invalidFeedback').css('display', 'none');
-      $('#textNotifierPadding').removeClass('pt-5');
-      msgFlash("Multiple time offering events saved!", "success");
+  });  
+
+  // Check if the start time is after the end time
+  for(let i = 0; i < startTimeInputs.length; i++){
+    if(startTimeInputs[i].value < endTimeInputs[i].value){
+      $(startTimeInputs[i]).removeClass('border-red');
+      $(endTimeInputs[i]).removeClass('border-red');
+    } else {
+      $(startTimeInputs[i]).addClass('border-red');
+      $(endTimeInputs[i]).addClass('border-red');
+      hasValidTimes = false;
     }
-  });
-//build multi-event table
-function storeMultipleOfferingEventAttributes() {
-    let entries = [];
-    $(".extraSlots").children().each(function(index, element) {
-        let rowData = $.map($(element).find("input"), (el) => $(el).val());
-
-        entries.push({
-            eventName: rowData[0],
-            eventDate: rowData[1],
-            startTime: rowData[2],
-            endTime: rowData[3]
-        });
+  }
+
+  // Check if there are duplicate event offerings
+  let eventListings = {};
+  for(let i = 0; i < eventOfferings.length; i++){
+    let eventName = eventNameInputs[i].value
+    let date = datePickerInputs[i].value.trim()
+    let startTime = startTimeInputs[i].value
+    let eventListing = JSON.stringify([eventName, date, startTime])
+
+    if (eventListing in eventListings){ // If we've seen this event before mark this event and the previous as duplicates
+      hasDuplicateListings = true
+      $(eventOfferings[i]).addClass('border-red');
+      $(eventOfferings[eventListings[eventListing]]).addClass('border-red')
+    } else { // If we haven't seen this event before
+      $(eventOfferings[i]).removeClass('border-red');
+      eventListings[eventListing] = i
+    }
+  }
+
+  if (isEmpty){
```

Comment:
Some of the code in this conditional and following conditionals seem to be redundant. Can we potentially create a new function that has a parameter that can be passed in to `('.invalidFeedback').text()` since the rest is identical.